### PR TITLE
140043 - BESS Cleanup Web Job and Logging Config

### PR DESCRIPTION
**Pull request by ratnas104142**
This PR consists changes of [PBI 140043](https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_workitems/edit/140043) -
- Code changes for cleanup webjob for BESS
- Unit Test
- Build, publish and deploy BESS cleanup webjob project

### CODE REVIEW

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs [hunk 1]
```diff
@@ -728,7 +728,31 @@ public enum EventIds
         /// <summary>
         /// 809144 - PRODUCT.TXT file and INFO folder delete operation failed
         /// </summary>
-        BessProductTxtAndInfoFolderDeleteFailed = 809144
+        BessProductTxtAndInfoFolderDeleteFailed = 809144,
+        /// <summary>
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs:732]:
```suggestion
        BessProductTxtAndInfoFolderDeleteFailed = 809144,
        
        /// <summary>
```

File: UKHO.PeriodicOutputService/UKHO.BESS.CleanUpJob/Program.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+using Microsoft.Extensions.Logging.Configuration;
+using Serilog;
+using Serilog.Events;
+using UKHO.BESS.CleanUpJob;
+using UKHO.BESS.CleanUpJob.Configuration;
+using UKHO.BESS.CleanUpJob.Services;
+using UKHO.Logging.EventHubLogProvider;
+using UKHO.PeriodicOutputService.Common.Configuration;
+
+namespace UKHO.ExchangeSetService.CleanUpJob
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.BESS.CleanUpJob/Program.cs:21]:
```suggestion
namespace UKHO.BESS.CleanUpJob
```

File: UKHO.PeriodicOutputService/UKHO.BESS.CleanUpJob/Services/BespokeExchangeSetCleanUpService.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+        public BespokeExchangeSetCleanUpService(IConfiguration configuration, ILogger<BespokeExchangeSetCleanUpService> logger, IOptions<CleanUpConfiguration> cleanUpConfig, IFileSystem fileSystem)
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.cleanUpConfig = cleanUpConfig ?? throw new ArgumentNullException(nameof(cleanUpConfig));
+            this.fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+            homeDirectoryPath = configuration == null ? throw new ArgumentNullException(nameof(configuration))
+                                : Path.Combine(configuration["HOME"]!, configuration["BespokeFolderName"]!);
+        }
+        public async Task CleanUpHistoricFoldersAndFiles()
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.BESS.CleanUpJob/Services/BespokeExchangeSetCleanUpService.cs:26]:
```suggestion
        }
        
        public async Task CleanUpHistoricFoldersAndFiles()
```

File: UKHO.PeriodicOutputService/UKHO.BESS.CleanUpJob/BespokeExchangeSetCleanUpJob.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        private readonly ILogger<BespokeExchangeSetCleanUpJob> logger;
+        private readonly IBespokeExchangeSetCleanUpService bespokeExchangeSetCleanUpService;
+
+        public BespokeExchangeSetCleanUpJob(ILogger<BespokeExchangeSetCleanUpJob> logger, IBespokeExchangeSetCleanUpService bespokeExchangeSetCleanUpService)
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.bespokeExchangeSetCleanUpService = bespokeExchangeSetCleanUpService ?? throw new ArgumentNullException(nameof(bespokeExchangeSetCleanUpService));
+        }
+        public async Task ProcessCleanUp()
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.BESS.CleanUpJob/BespokeExchangeSetCleanUpJob.cs:20]:
```suggestion
        }
        
        public async Task ProcessCleanUp()
```

File: UKHO.PeriodicOutputService/UKHO.BESS.CleanUpJob/settings.job [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "schedule": "0 15 16 * * *"
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.BESS.CleanUpJob/settings.job:2]: This might need to revert to get the cleanup done in non working hours.

**Review by harshal11869 — CHANGES_REQUESTED** [review]
Few minor comments

File: UKHO.PeriodicOutputService/UKHO.BESS.CleanUpJob/settings.job (hunk 5 above)
Comment (ratnas104142) [reply UKHO.PeriodicOutputService/UKHO.BESS.CleanUpJob/settings.job:2]: This cron expression changed some time ago for dev testing purpose. Will revert it back to 0 0 3 * * * once dev testing is completed. Thanks!

Comment (harshal11869) [reply UKHO.PeriodicOutputService/UKHO.BESS.CleanUpJob/settings.job]: Ok

**Review by harshal11869 — CHANGES_REQUESTED** [review]

**Review by harshal11869 — APPROVED** [review]
Looks Good.

**Review by Nadeem14456 — APPROVED** [review]
LGTM.

**Review by DanOwenUKHO — APPROVED** [review]
LGTM

**Pull request merged**